api/envoy/extensions/load_balancing_policies/override_host/v3/override_host.proto

```diff
@@ -0,0 +1,72 @@
+syntax = "proto3";
+
+package envoy.extensions.load_balancing_policies.override_host.v3;
+
+import "envoy/config/cluster/v3/cluster.proto";
+import "envoy/type/metadata/v3/metadata.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.load_balancing_policies.override_host.v3";
+option java_outer_classname = "OverrideHostProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/load_balancing_policies/override_host/v3;override_hostv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Override Host Load Balancing Policy]
+// [#extension: envoy.load_balancing_policies.override_host]
+
+// Configuration for the Override Host Load Balancing policy.
+//
+// This policy allows endpoint picking to be implemented in an external to Envoy
+// component. While it can work with a built in HTTP filter it is primarily intended to
+// work in conjunction with the ext_proc or WASM HTTP filters.
+//
+// The ext_proc or WASM filter must implement k8s proposal for AI gateway inferences extensions
```

Comment:
I don't think this is actually required.  The config is now general-purpose enough that the LB policy really just requires that a header or dynamic metadata be set by *something*.  It doesn't need to be ext_proc or WASM, and it doesn't need to have anything to do with inference gateway.

I suggest saying something like "One example of where this is useful is in conjunction with an ext_proc or WASM filter that implements the k8s proposal for ...".

Reply:
Is stateful session doing essentially the same thing but with a request itself instead of a look-up? Can you use that HTTP filter with this policy?

Reply:
There was a *lot* of earlier discussion about exactly this point.  See https://docs.google.com/document/d/1JYxILl-QJbcArmcCCQzvPIpk9cSogXAzYGnSsWWTZZE/edit?usp=sharing for a summary of the options we considered.

Basically, I would like to ultimately change the stateful session filter to use this LB policy instead of using the override_host mechanism that is currently built into the cluster, but we don't currently have consensus on a plan to deprecate the existing mechanism.

Reply:
OK, that's plenty of discussion. It makes sense to keep it policy-driven rather than done in a downstream filter code.

Reply:
> OK, that's plenty of discussion. It makes sense to keep it policy-driven rather than done in a downstream filter code.

cc @kyessenov 

Ideally, I hope we can replace the stateful session in the future with the new LB policy. But per our previous discussion, it's impossible to remove the dependency to the downstream filter.

Reply:
Done